lib/cmr/graphql_client.rb

```diff
@@ -0,0 +1,177 @@
+      sslPolicy = {}
+      sslPolicy['sslEnabled'] = payload[:sslPolicy][:sslEnabled] == '1' ? true : false
+      sslPolicy['sslCertificate'] = payload[:sslPolicy][:sslCertificate]
+      variables['sslPolicy'] = sslPolicy
+      query = {}
+      query['variables'] = variables
+      query['query'] = "mutation UpdateProviderPolicy(
+                          $providerId: String!
+                          $endpoint: String!
+                          $retryAttempts: Int!
+                          $retryWaitTime: Int!
+                          $orderingSuspendedUntilDate: Timestamp
+                          $maxItemsPerOrder: Int
+                          $overrideNotifyEnabled: Boolean
+                          $referenceProps: String
+                          $sslPolicy: SslPolicyInput!
+                        ) {
+                            updateProviderPolicy(
+                              endpoint: $endpoint
+                              providerId: $providerId
+                              retryAttempts: $retryAttempts
+                              retryWaitTime: $retryWaitTime
+                              orderingSuspendedUntilDate: $orderingSuspendedUntilDate
+                              maxItemsPerOrder: $maxItemsPerOrder
+                              overrideNotifyEnabled: $overrideNotifyEnabled
+                              referenceProps: $referenceProps
+                              sslPolicy: $sslPolicy
+                          ) {
+                            id
+                          }
+                        }"
+      url = '/ordering/api'
+      headers = { 'Content-Type' => 'application/json' }
+      headers = headers.merge(authorization_header(token))
+      post(url, query.to_json, headers)
+    end
+
+    def test_endpoint_connection(token, provider_id)
+      variables = {}
+      variables['providerId'] = provider_id
+      query = {}
+      query['variables'] = variables
+      query['query'] = "mutation TestProviderConnection($providerId: String!) {
+                          testProviderConnection(providerId: $providerId) {
+                            status
+                          }
+                        }"
+      url = '/ordering/api'
+      headers = { 'Content-Type' => 'application/json' }
+      headers = headers.merge(authorization_header(token))
+      post(url, query.to_json, headers)
+    end
+
+    def remove_provider_policy(token, provider_id)
+      variables = {}
+      variables['providerId'] = provider_id
+      query = {}
+      query['variables'] = variables
+      query['query'] = "mutation DeleteProviderPolicy($providerId: String!) {
+                          deleteProviderPolicy(providerId: $providerId)
+                        }"
+      url = '/ordering/api'
+      headers = { 'Content-Type' => 'application/json' }
+      headers = headers.merge(authorization_header(token))
+      post(url, query.to_json, headers)
+    end
+  end
+end
```

Comment:
add the following to .editorconfig (which already exists) under the `[*]` section
`end_of_line = lf
insert_final_newline = true
trim_trailing_whitespace = true`
and never have to see these again

Reply:
Done